### PR TITLE
Openfinex api mock

### LINKED ISSUE #85: Mock the calls to OpenFinex API in polkadex gateway to allow clean unit testing
The polkadex gateway needs an interface to the OpenFinex API that allows it to be mocked for unit testing. This might require to put the polkadex gateway functions in a struct itself, that holds an OpenFinex API trait object.

The current state means, we cannot have passing unit tests and 'real' calls into OpenFinex at the same time, which is sub-optimal to say the least.

### PR DESCRIPTION
**Pull request by haerdib**
resolves #85

It is not a perfect solution, but it works. I've opened a seperate issue for cleaning up the code: #111

### CODE REVIEW

**Review by Gauthamastro — APPROVED** [review]

File: enclave/src/polkadex_gateway.rs [hunk 1]
```diff
@@ -114,42 +114,85 @@ impl PolkaDexGatewayCallback for PolkaDexGatewayCallbackImpl {
...
-    match order.order_type {
-        OrderType::LIMIT => {
-            if order.quantity == 0 as Balance {
-                error!("Limit Order quantity Zero");
-                return Err(GatewayError::QuantityZeroInLimitOrder);
+/// All sendings to the openfinex server should go through
+/// this gateway. Necessary to mock unit tests
+pub struct OpenfinexPolkaDexGateway<B: OpenFinexApi> {
+    openfinex_api: B
+}
```
Comment (murerfel) [inline enclave/src/polkadex_gateway.rs:121]: This is interesting, I didn't know you could compose a struct like this, using a trait as type parameter and then use it as trait object

**Comment by murerfel** [comment]
Good job, I think this is exactly what I had in mind when I wrote the issue 👍

**Review by murerfel — APPROVED** [review]
Ah I added a comment, again instead of reviewing. my bad! This looks good to me - I'll have to check out the kind of composition you did for the OpenFinexAPI 😄

File: enclave/src/polkadex_gateway.rs (hunk 1 above)
Comment (haerdib) [reply enclave/src/polkadex_gateway.rs:121]: It was my first time using it like this aswell. I copied from the substrate code (top_pool has a lot of these kind of implementations: https://github.com/Polkadex-Substrate/polkadexTEE-worker/blob/master/enclave/src/top_pool/pool.rs#L130-L135)

**Pull request merged**